test/lib/cql_test_env.cc

```diff
@@ -644,10 +649,21 @@ class single_node_cql_env : public cql_test_env {
                 return make_ready_future<>();
             }).get();
 
+            uint16_t port = 7000;
+            if (cfg_in.ms_listen) {
+               std::random_device rd;
```

Comment:
Why not use the test random device for reproducibility?
Is the seed resolution too low?

Also, the comment below isn't accurate with this change.
It can be deleted, and replaced with a comment explaining this randomization of the listen port.

Reply:
It might also be a good idea to add a command-line option to set the listen address and port so that test.py or the developer could manage those in a central way.
But that can be done in a follow up.

Reply:
I am not aware of the "test random device" you mentioned. We just need a random port. We do not need them to be reproducible.

Reply:
`seastar::testing::local_random_engine` it's seeded by default using `std::random_device` and that seed is printed to the log so that one can use the `--random-seed` command line option to reproduce the run.

You don't need reproducibility until it breaks... :)
Anyway, it doesn't cost anything to use the `local_random_engine` so we should use it by convention in tests, just in case.

Reply:
If it is free, why not. I meant we do not need to reproduce the exact port number which was used. As soon as the port chosen by the test is already taken. We will see in the log the port is taken and scylla refuses to start. It can be someone else who is using the port, not necessarily other instance of the same test, reproducing the port number is not going to help in this case.

Reply:
It is not free.

ld.lld: error: undefined symbol: seastar::testing::local_random_engine

https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/5083/artifact/output-build-x86_64.txt

Reply:
e.g, ld.lld: error: undefined symbol: seastar::testing::local_random_engine

Reply:
@bhalevy ^^^

Reply:
It's probably tests in `tests_not_using_seastar_test_framework` (See configure.py)
Any test there that uses cql_test_env needs to be moved out of that list.

Reply:
It is a mess to deal with that especially there is a oss and ent difference  in play.  I will revert to my old version which does not use seastar::testing::local_random_engine.

Reply:
ok, we can change it later on if needed